### DISCUSSION TITLE
Tag 'html' marked with attribute 'data-ampdevmode' AMP

### DISCUSSION THREAD

**Discussion by birshu123** [vercel/next.js#15317] [Help]
Dear Friends,

I an new to nextjs. I am using next to generate amp pages for my website.

But is Amp validator it gives error: Tag 'html' marked with attribute 'data-ampdevmode'

Can anyone please help what to do?

Test link: https://bengali.nenow.in/assam/the-boy-killed-his-father-over-a-land-dispute-in-assam/amp

-Birshu

**Comment by sebastianbenz** [comment]
This is working as intended. Next.js adds custom JS during development which would be invalid AMP. `data-ampdevmode` allows Next.js to filter validation errors caused by the custom JS. You can ignore this. In prod builds the attribute won't be present.
